xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/test/java/org/xwiki/search/solr/internal/EmbeddedSolrInitializationTest.java

```diff
@@ -209,4 +212,33 @@ void testInstantiationWhenDoesNotExist() throws Exception
 
         getInstanceAndAssertHomeDirectory(newHome);
     }
+    
+    /**
+     * This looks for a regression where the file.seperator on windows was not properly escaped. 
+     * in the ""xwiki\store\solr\search\core.properties" file. 
+     */
+    @Test
+    void testCacheHomeProperlyEscapedOnWindows() throws Exception {
+        String newHome = new File(this.permanentDirectory, "doesNotExist").getAbsolutePath();
+        when(this.mockXWikiProperties.getProperty(eq(SOLRHOME_PROPERTY), anyString())).thenReturn(newHome);
+        
+        Solr instance = this.componentManager.getInstance(Solr.class, EmbeddedSolr.TYPE);
+        EmbeddedSolr implementation = ((EmbeddedSolr) instance);
+        CoreContainer container = implementation.getContainer();
+        SolrCore core = container.getCore(container.getLoadedCoreNames().get(0));
+        File coreBaseDirectory = new File(container.getSolrHome(), core.getName());       
+        
+        File file = new File(coreBaseDirectory,"core.properties");
+        assertTrue(file.exists(), "Couldn't find solr cache properties file: "+file.toString());
+        Properties properties = new Properties();
+        try( BufferedReader in = new BufferedReader(new FileReader(file, StandardCharsets.UTF_8)); ){
+        	properties.load(in);
+        }
+        String dataDir = properties.getProperty("dataDir");
+        assertNotNull(dataDir, "dataDir property from properties file was null: "+file.toString());
+        String fileSeperator = System.getProperty("file.separator");
```

Comment:
Just a codestyle detail: you may want to use the more obvious File.separator (or File.separatorChar) for this.

Reply:
Has that always been there? There's not a "since" on it.  I didn't even know that was there in the files class. Yes much more obvious and convenient.

Reply:
Not sure honesty, but that's how I would interpret the lack of `@since` too.